compiler/parser/parser.cpp

```diff
@@ -328,7 +328,23 @@ static void parseCommandLineFiles() {
   }
 
   while ((inputFileName = nthFilename(fileNum++))) {
-    if (isChplSource(inputFileName)) {
+    if (isChplSource(inputFileName))
+    {
+      /*
+      Ensure that all the files parsed don't exceed the
+      (NAME_MAX - 16) i.e. 239 bytes Limits
+      */
+      const size_t maxFileName = NAME_MAX - 16;
+      if (strlen(inputFileName) > maxFileName)
+      {
+        char reducedFileName[maxFileName];
+        // sprintf fileName to allowed length buffer
+        strncpy(reducedFileName, inputFileName, maxFileName);
+        // error message to print between filename and filename limit
+        const char *errorMessage = "%s, filename is longer than maximum allowed length of %d\n";
+        // throwr error will concatenated messages
+        USR_FATAL(errorMessage, reducedFileName, maxFileName);
```

Comment:
@king-11 : What's the rationale for reducing the filename rather than just printing out the whole thing?  It seems simpler and more accurate for the user if the code simply did:

```c
USR_FATAL(errorMessage, inputFileName, maxFileName);
```

Guessing:  If the rationale is so that the output is predictable in testing, then I think we should have the .precomp file also generate the .good file.  It seems more important to print accurate information to the user than to truncate the filename without explanation.

Repeating some comments [from before](https://github.com/chapel-lang/chapel/pull/17559#issuecomment-827247484):

* I think the comment should rationalize the choice of 16 so that people don't wonder about it in the future.  E.g., "This is to give us some breathing room so that we can create variations on the input filename, such as `baseName.tmp.obj` without exceeding `NAME_MAX`".
* "i.e.," means "that is", but there's no guarantee that NAME_MAX will have the value of 256.  So instead, I might say something like "e.g., 239 bytes on [name of some configuration where you found it to be 256."

Reply:
Oops, the precomp already generates the .good, so that guess was wrong.

Reply:
Corrected all three points